core/services/job/models.go

```diff
@@ -850,47 +850,38 @@ type LiquidityBalancerSpec struct {
 }
 
 type WorkflowSpec struct {
-	ID int32 `toml:"-"`
-	// TODO it may be possible to compute the workflow id from the hash(yaml, owner, name) and remove this field
-	WorkflowID    string    `toml:"workflowId"` // globally unique identifier for the workflow, specified by the user
-	Workflow      string    `toml:"workflow"`
-	WorkflowOwner string    `toml:"workflowOwner"` // hex string representation of 20 bytes
-	WorkflowName  string    `toml:"workflowName"`  // 10 byte plain text name
+	ID       int32  `toml:"-"`
+	Workflow string `toml:"workflow"` // the yaml representation of the workflow
+	// fields derived from the yaml spec, used for indexing the database
+	// note: i tried to make these private, but translating them to the database seems to require them to be public
+	WorkflowID    string    `toml:"-" db:"workflow_id"`    // Derived. Do not modify. the CID of the workflow.
+	WorkflowOwner string    `toml:"-" db:"workflow_owner"` // Derived. Do not modify. the owner of the workflow.
+	WorkflowName  string    `toml:"-" db:"workflow_name"`  // Derived. Do not modify. the name of the workflow.
 	CreatedAt     time.Time `toml:"-"`
 	UpdatedAt     time.Time `toml:"-"`
 }
 
 var (
-	ErrInvalidWorkflowID    = errors.New("invalid workflow id")
-	ErrInvalidWorkflowOwner = errors.New("invalid workflow owner")
-	ErrInvalidWorkflowName  = errors.New("invalid workflow name")
+	ErrInvalidWorkflowID       = errors.New("invalid workflow id")
+	ErrInvalidWorkflowYAMLSpec = errors.New("invalid workflow yaml spec")
 )
 
 const (
-	workflowIDLen = 64 // conveniently the same length as a sha256 hash
-	// owner and name are constrained the onchain representation in [github.com/smartcontractkit/chainlink-common/blob/main/pkg/capabilities/consensus/ocr3/types/Metadata]
-	workflowOwnerLen = 40 // hex string representation of 20 bytes
-	workflowNameLen  = 10 // plain text name
+	workflowIDLen = 64 // sha256 hash
 )
 
-// Validate checks the length of the workflow id, owner and name
-// that latter two are constrained by the onchain representation in [github.com/smartcontractkit/chainlink-common/blob/main/pkg/capabilities/consensus/ocr3/types/Metadata]
+// Validate checks the workflow spec for correctness
 func (w *WorkflowSpec) Validate() error {
-	if len(w.WorkflowID) != workflowIDLen {
-		return fmt.Errorf("%w: incorrect length for id %s: expected %d, got %d", ErrInvalidWorkflowID, w.WorkflowID, workflowIDLen, len(w.WorkflowID))
-	}
-
-	w.WorkflowOwner = strings.TrimPrefix(w.WorkflowOwner, "0x")
-	_, err := hex.DecodeString(w.WorkflowOwner)
+	s, err := pkgworkflows.ParseWorkflowSpecYaml(w.Workflow)
 	if err != nil {
-		return fmt.Errorf("%w: expected hex encoding got %s: %w", ErrInvalidWorkflowOwner, w.WorkflowOwner, err)
-	}
-	if len(w.WorkflowOwner) != workflowOwnerLen {
```

Comment:
Why aren't we validating Owner length any more?

Reply:
it's happening in the json schema validation of the YAML here https://github.com/smartcontractkit/chainlink-common/blob/c1110605a9906fcc64f0c105810c1835d6702075/pkg/workflows/models_yaml.go#L67

this enable CLO to reuse the validation